### PR TITLE
feat: [M3-7351] - AGLB Configurations Add Route Drawer and other refinements

### PR DESCRIPTION
**Pull request by bnussman-akamai**
## Description 📝

- Updates AGLB Configurations to match UX mockups
- Adds Add Route drawer where you can
  - Add an existing route
  - Create a new route
- Adds `RouteSelect` component
- Fixes some position/styling
- Componentizes the Beta Regions 

## Preview 📷

| Before  | After   |
| ------- | ------- |
| Screenshot 2023-10-30 at 6 14 21 PM | Screenshot 2023-10-30 at 6 13 50 PM |
| Screenshot 2023-10-31 at 9 58 41 AM | Screenshot 2023-10-31 at 9 58 26 AM |

### Prerequisites
- Turn on the MSW 

### Verification steps 
- Go to http://localhost:3000/loadbalancers/0/configurations
- Check the UI for styling issues
- Check functionality of the `Add Route` button and the corresponding drawer (We can only test so much with the MSW)

## As an Author I have considered 🤔

- [x] 👀 Doing a self review
- [x] ❔ Our [contribution guidelines](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)
- [x] 🤏 Splitting feature into small PRs
- [x] ➕ Adding a changeset
- [x] 🧪 Providing/Improving test coverage
- [x] 🔐 Removing all sensitive information from the code and PR description
- [x] 🚩 Using a feature flag to protect the release
- [x] 👣 Providing comprehensive reproduction steps
- [x] 📑 Providing or updating our documentation
- [x] 🕛 Scheduling a pair reviewing session
- [x] 📱 Providing mobile support
- [x] ♿  Providing accessibility support

### CODE REVIEW

File: packages/manager/src/features/LoadBalancers/LoadBalancerDetail/Routes/RoutesTable.tsx [hunk 1]
```diff
@@ -204,50 +187,6 @@ export const RoutesTable = ({ configuredRoutes }: Props) => {
 
   return (
     <>
-      <Stack
```
Comment (bnussman-akamai) [inline packages/manager/src/features/LoadBalancers/LoadBalancerDetail/Routes/RoutesTable.tsx:207]: Moved this logic outside of this component so we can re-use it between the Routes tab and the configurations tab

File: packages/manager/src/features/LoadBalancers/LoadBalancerDetail/Routes/RoutesTable.tsx [hunk 2]
```diff
@@ -135,20 +125,13 @@ export const RoutesTable = ({ configuredRoutes }: Props) => {
...
               actionText="Add Rule"
               onClick={() => onAddRule(route)}
             />
-            {/**
-             * TODO: AGLB: The Action menu behavior should be implemented in future AGLB tickets.
-             */}
             <ActionMenu
               actionsList={[
                 { onClick: () => onEditRoute(route), title: 'Edit' },
-                { onClick: () => null, title: 'Clone Route' },
```
Comment (bnussman-akamai) [inline packages/manager/src/features/LoadBalancers/LoadBalancerDetail/Routes/RoutesTable.tsx:151]: We're not going to do cloning for beta

File: packages/manager/src/features/LoadBalancers/LoadBalancerDetail/Configurations/ConfigurationAccordion.tsx
Comment (mjac0bs) [inline packages/manager/src/features/LoadBalancers/LoadBalancerDetail/Configurations/ConfigurationAccordion.tsx:1]: I don't know that this PR is necessarily the right place for it, but do we need some error handling on the configurations tab? If we block network requests to the configurations GET endpoint, we're currently lacking user feedback when those configs fail to load. 

https://github.com/linode/manager/assets/114685994/eeb02713-3802-4676-a580-73df5f7d923c

**Review by mjac0bs — COMMENTED** [review]
I tested Add Route drawer functionality and confirmed that adding a new route sends a POST request and form validation works for both adding a new route and adding an existing route.

I haven't seen any major styling issues, but a have commented a couple suggestions related to service target table and the config tab. I'll take another glance through the UI to finish confirming styling after this initial feedback.

I also don't know the best spot for this feedback and know copy is being finalized anyway, but I don't want to accidentally miss these if this copy doesn't change much - Configure tab > Apply Certificate drawer:
Screenshot 2023-11-01 at 12 48 49 PM

**Comment by bnussman-akamai** [comment]
Updated copy will be coming very soon! @mjac0bs 


I think I addressed your feedback! Let me know if I missed anything!

File: packages/manager/.changeset/pr-9853-upcoming-features-1698871857708.md
Comment (mjac0bs) [inline packages/manager/.changeset/pr-9853-upcoming-features-1698871857708.md:1]: We already had a changeset for the manager package, so now this is a duplicate!

File: packages/manager/src/features/LoadBalancers/LoadBalancerDetail/Configurations/ConfigurationAccordion.tsx [hunk 3]
```diff
@@ -29,11 +29,30 @@ interface Props {
   loadbalancerId: number;
 }
 
+function getConfigurationPayloadFromConfiguration(
```
Comment (cpathipa) [inline packages/manager/src/features/LoadBalancers/LoadBalancerDetail/Configurations/ConfigurationAccordion.tsx:32]: Can we name it `buildConfigurationPayload` / `generateConfigPayload` ? `getConfigurationPayloadFromConfiguration` is a bit confusing.

**Review by cpathipa — APPROVED** [review]
Overall functionality LGTM!  left couple of non blocking comments.

**Review by mjac0bs — APPROVED** [review]
Add Routes drawer still looks good, error state for Configurations tab looks good, and thanks for adding the ID column to Service Targets (and ID elsewhere, it looks like).

Non-blocking: one more spot in the mocks where we have an ID -- the certificate ID and created timestamp -- can we add that here now?:
image

Worth asking because there are are little changes in this PR -- are there any additional unit or e2e updates we want to make to check for the IDs displaying in the UI, etc?

**Pull request merged**